admin/server/projects.go

```diff
@@ -711,31 +711,44 @@ func (s *Server) RemoveProjectMember(ctx context.Context, req *adminv1.RemovePro
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
 
-	claims := auth.GetClaims(ctx)
-	if !claims.ProjectPermissions(ctx, proj.OrganizationID, proj.ID).ManageProjectMembers {
-		return nil, status.Error(codes.PermissionDenied, "not allowed to remove project members")
-	}
-
 	user, err := s.admin.DB.FindUserByEmail(ctx, req.Email)
 	if err != nil {
 		if !errors.Is(err, database.ErrNotFound) {
 			return nil, status.Error(codes.Internal, err.Error())
 		}
-		// check if there is a pending invite
+
+		// Only admins can remove pending invites.
+		// NOTE: If we change invites to accept/decline (instead of auto-accept on signup), we need to revisit this.
+		claims := auth.GetClaims(ctx)
+		if !claims.ProjectPermissions(ctx, proj.OrganizationID, proj.ID).ManageProjectMembers {
+			return nil, status.Error(codes.PermissionDenied, "not allowed to remove project members")
+		}
+
+		// Check if there is a pending invite
 		invite, err := s.admin.DB.FindProjectInvite(ctx, proj.ID, req.Email)
 		if err != nil {
 			if errors.Is(err, database.ErrNotFound) {
 				return nil, status.Error(codes.InvalidArgument, "user not found")
 			}
 			return nil, status.Error(codes.Internal, err.Error())
 		}
+
 		err = s.admin.DB.DeleteProjectInvite(ctx, invite.ID)
 		if err != nil {
 			return nil, status.Error(codes.Internal, err.Error())
 		}
 		return &adminv1.RemoveProjectMemberResponse{}, nil
 	}
 
+	// The caller must either have ManageProjectMembers permission or be the user being removed.
+	claims := auth.GetClaims(ctx)
+	allow := false
+	allow = allow || claims.ProjectPermissions(ctx, proj.OrganizationID, proj.ID).ManageProjectMembers
+	allow = allow || (claims.OwnerType() == auth.OwnerTypeUser && claims.OwnerID() == user.ID)
```

Comment:
```suggestion
	allow := claims.ProjectPermissions(ctx, proj.OrganizationID, proj.ID).ManageProjectMembers || (claims.OwnerType() == auth.OwnerTypeUser && claims.OwnerID() == user.ID)
```

Reply:
Google's Go style guide recommends breaking complex conditionals into multiple lines: https://google.github.io/styleguide/go/guide#maintainability.

I refactored it to use separate vars instead:
```go
	isManager := claims.OrganizationPermissions(ctx, org.ID).ManageOrgMembers
	isSelf := claims.OwnerType() == auth.OwnerTypeUser && claims.OwnerID() == user.ID
	if !isManager && !isSelf {
```

Reply:
ok